TimerBroker Initial Commit

holtchesley opened the pull request:
Initial Commit of TimerBroker, which handles timers for cluster members.
Members must implement the timer_trip method.

Commit:
TimerBroker Initial Commit

Initial Commit of TimerBroker, which handles timers for cluster members.
Members must implement the timer_trip method.

WesleyAC commented on broker/timer.py:
```diff
@@ -0,0 +1,42 @@
+        elif step[0] == 'Delay':
+            delay = step[1]
+            timer_bumped = self.timer_queue.pop(0)
+            self.timer_queue = self.timer_queue[0:-delay] + [timer_bumped] + self.timer_queue[-delay:]
```
Pretty sure you can do `self.timer_queue.insert(delay, timer_bumped)`

Commit:
Remove timer_ids from TimerBroker

Having multiple different timer_ids involved in the timers/callbacks
is unnecessary, and complicates implementations. timer_id usage should
be removed.

Commit:
Prefer built in python insert

Commit:
Remove timer_id ref from public API of TimerBroker
